test/replication-luatest/gh_7318_box_wait_limbo_acked_test.lua

```diff
@@ -0,0 +1,433 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local proxy = require('luatest.replica_proxy')
+local server = require('luatest.server')
+
+local g = t.group('cover-box-wait-limbo-acked')
+--
+-- gh-7318:
+-- Cover box_wait_limbo_acked with tests.
+--
+local wait_timeout = 10
+
+local function wait_pair_sync(server1, server2)
+    -- Without retrying it fails sometimes when vclocks are empty and both
+    -- instances are in 'connect' state instead of 'follow'.
+    t.helpers.retrying({timeout = wait_timeout}, function()
+        server1:wait_for_vclock_of(server2)
+        server2:wait_for_vclock_of(server1)
+        server1:assert_follows_upstream(server2:get_instance_id())
+        server2:assert_follows_upstream(server1:get_instance_id())
+    end)
+end
+
+local function check_wal_is_blocked_f()
+    if not box.error.injection.get('ERRINJ_WAL_DELAY') then
+        error('WAL still is not paused')
+    end
+end
+
+local function server_wait_wal_is_blocked(server)
+    t.helpers.retrying({timeout = wait_timeout}, server.exec, server,
```

Comment:
Nit: you may retry inside `server.exec`. This way there will be no need to retry many exec calls. Instead, all the retrying will be done inside one exec call.
For example, `check_wal_is_blocked_f` may be turned into `wait_wal_is_blocked_f`.

Same for other `check_..._f` and `wait_...` pairs.